crates/milli/src/search/new/bucket_sort.rs

```diff
@@ -159,7 +161,12 @@ pub fn bucket_sort<'ctx, Q: RankingRuleQueryTrait>(
         };
     }
 
-    while valid_docids.len() < length {
+    let max_total_hits = max_total_hits.unwrap_or(usize::MAX);
```

Comment:
to avoid accidental full scans of the DB should `max_total_hits` somehow not be set, I advise we only take into account `exhaustive_number_hits` when `max_total_hits.is_some()`.

A nice way to express that could be the following:

```rust
let max_len_to_evaluate = match (max_total_hits, exhaustive_number_hits && ranking_score_threshold.is_some()) {
    (Some(max_total_hits), true) => max_total_hits,
    _ => length
};

while valid_docids.len() < max_len_to_evaluate {
// ...
}